app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/index.tsx

```diff
@@ -0,0 +1,218 @@
+  &.MuiSvgIcon-root {
+    width: 24px;
+    height: 24px;
+  }
+`;
+
+const SelectColumnSearchInput = styled.input`
+  margin-right: 5px;
+  border: none;
+  border-bottom: 1px solid transparent;
+  margin-bottom: 5px;
+  &:focus {
+    border-bottom: 1px solid ${grey[700]};
+    outline: none;
+  }
+`;
+
+const SelectColumnWrapper = styled(Box)`
+  height: 90%;
+`;
+
+const SelectColumnInnerWrapper = styled(Box)`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 10px;
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+const SelectColumnSearchBox = styled(Box)`
+  position: relative;
+  display: flex;
+`;
+
+const SearchIconButton = styled(IconButton)`
+  padding: 5px 0px 5px 5px;
+
+  &.MuiIconButton-root:hover {
+    background-color: transparent;
+  }
+  & .MuiTouchRipple-root {
+    display: none;
+  }
+`;
+
+export default function({
+  transformationDataType,
+  selectedColumnsCount,
+  columnsList,
+  setSelectedColumns,
+  dataQuality,
+  transformationName,
+  selectedColumns,
+}: ISelectColumnsListProps) {
+  const [columns, setColumns] = useState<IHeaderNamesList[]>(columnsList);
+  const [isSingleSelection, setIsSingleSelection] = useState<boolean>(true);
+  const ref = useRef(null);
+
+  const columnsAsPerType = getColumnsSupportedType(transformationDataType, columnsList);
+  const filteredColumnsOnType = getFilteredColumn(transformationDataType, columnsList);
+
+  useEffect(() => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN?.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName
+    );
+
+    if (multiSelect.length) {
+      setIsSingleSelection(false);
+    }
+    setColumns(filteredColumnsOnType);
+  }, []);
+
+  const onSingleSelection = (column: IHeaderNamesList) => {
+    setSelectedColumns([column]);
+  };
+
+  const onMultipleSelection = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    column: IHeaderNamesList
+  ) => {
+    if (event.target.checked) {
+      setSelectedColumns((prev) => [...prev, column]);
+    } else {
+      const indexOfUnchecked = selectedColumns.findIndex(
+        (columnDetail) => columnDetail.label === column.label
+      );
+      if (indexOfUnchecked > -1) {
+        setSelectedColumns(() => selectedColumns.filter((_, index) => index !== indexOfUnchecked));
+      }
+    }
+  };
+
+  const handleDisableCheckbox = () => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName && functionDetail.isMoreThanTwo
+    );
+    if (selectedColumns?.length === 0 || selectedColumns?.length < 2) {
+      return false;
+    } else if (multiSelect.length) {
```

Comment:
i don't think we need else if and else here
if(a||b) return false
if(c) return false
return true

also we might not need curly braces or blocks for single liners unless the google inbuild linter is formatting it
this should suffice as the execution won't go to the next line if something is returned already

Reply:
Done